tests/tests/test-staking/test-staking-locks.ts

```diff
@@ -0,0 +1,390 @@
+      randomAccount.address
+    );
+    expect(delegatorState.unwrap().delegations.length).to.be.equal(
+      Number(maxDelegationsPerDelegator),
+      "Missing delegation"
+    );
+    // Only 1 over the 2 delegations has been revoked
+    const locks = await context.polkadotApi.query.balances.locks(randomAccount.address);
+    expect(locks.length).to.be.equal(1, "Missing lock");
+    expect(locks[0].amount.toBigInt()).to.be.equal(DELEGATE_AMOUNT * maxDelegationsPerDelegator);
+    expect(locks[0].id.toHuman().toString()).to.be.equal("stkngdel");
+  });
+});
+
+describeDevMoonbeam("Staking - Locks", (context) => {
+  const randomAccount = generateKeyingPair();
+
+  before("Setup candidate & delegations", async function () {
+    await context.createBlock([
+      context.polkadotApi.tx.balances.transfer(
+        randomAccount.address,
+        MIN_GLMR_STAKING * 2n + 1n * GLMR
+      ),
+      context.polkadotApi.tx.parachainStaking
+        .joinCandidates(MIN_GLMR_STAKING, 1)
+        .signAsync(baltathar),
+    ]);
+
+    let nonce = await context.web3.eth.getTransactionCount(randomAccount.address);
+    await context.createBlock([
+      context.polkadotApi.tx.parachainStaking
+        .delegate(alith.address, DELEGATE_AMOUNT, 10, 10)
+        .signAsync(randomAccount, { nonce: nonce++ }),
+      context.polkadotApi.tx.parachainStaking
+        .delegate(baltathar.address, DELEGATE_AMOUNT, 10, 10)
+        .signAsync(randomAccount, { nonce: nonce++ }),
+    ]);
+  });
+
+  it("should not be created for additional delegations", async function () {
+    const locks = await context.polkadotApi.query.balances.locks(randomAccount.address);
+    expect(locks.length).to.be.equal(
+      1,
+      `Unexpected number of locks: ${locks.map((l) => l.id.toHuman().toString()).join(` - `)}`
+    );
+  });
+
+  it("should increase for additional delegations", async function () {
+    const locks = await context.polkadotApi.query.balances.locks(randomAccount.address);
+    expect(locks[0].id.toHuman().toString()).to.be.equal("stkngdel");
+    expect(locks[0].amount.toBigInt(), `Unexpected amount for lock`).to.be.equal(
+      2n * DELEGATE_AMOUNT
+    );
+  });
+});
+
+describeDevMoonbeam("Staking - Locks", (context) => {
+  it("should be created when joining candidates", async function () {
+    await context.createBlock(
+      context.polkadotApi.tx.parachainStaking
+        .joinCandidates(MIN_GLMR_STAKING, 1)
+        .signAsync(baltathar)
+    );
+
+    const locks = await context.polkadotApi.query.balances.locks(baltathar.address);
+    expect(locks.length).to.be.equal(
+      1,
+      `Unexpected number of locks: ${locks.map((l) => l.id.toHuman().toString()).join(` - `)}`
+    );
+    expect(locks[0].amount.toBigInt()).to.be.equal(MIN_GLMR_STAKING);
+    expect(locks[0].id.toHuman().toString()).to.be.equal("stkngcol");
+  });
+});
+
+const testFeesForHotfixExtrinsicWithNumDelegators = async (
```

Comment:
@crystalin feel free to make suggestions or clean this up. I borrowed this pattern for calculating fees from `test-length-fees.ts`, which was there because I wasn't sure how else we could get the exact fees that were paid.

(same for `testFeesForHotfixExtrinsicWithNumCollators` below)

Reply:
The test is temporary, so not too worried about it. The `test-length-fees.ts` could use a good refactor when you improve though.